### ISSUE TITLE
Failed to determine anonymous class for PsiClass:ExtPlayer

### ISSUE THREAD

**Issue by MadMartian** [DemonWav/MinecraftDevIntelliJ#442]
* Minecraft Plugin: 2017.3-1.2.0
* IntelliJ: IC-173.4674.33
* Linux Kubuntu 16.04.4 LTS
* KDE Plasma: 5.8.9
* KDE Frameworks: 5.36.0
* Qt: 5.6.1

This issue occurs in IntelliJ when attempting to opening a specific JUnit class file, then syntax highlighting breaks down and if I close the file I am unable to open that specific unit test file again.

I have no clue what specifically caused this isssue, so I have added the entire contents of the trouble-file below.  This is part of a scripting engine I am working on that is independent of Minecraft, so it doesn't even need the plugin at all, but I guess somehow the facet got added to the module.

```
/*
 * Olivine Script - Copyright (c) 2017 extollIT Enterprises.
 */

package com.extollit.olivine.runtime;

import com.extollit.linalg.immutable.Vec3d;
import com.extollit.olivine.annotation.OlivineType;
import com.extollit.olivine.ast.OlivineProgram;
import com.extollit.olivine.ex.OlivineException;
import com.extollit.olivine.ex.morph.InvalidMorphException;
import com.extollit.olivine.ex.morph.MalformedMorphException;
import com.extollit.olivine.ex.morph.MorphExistsException;
import com.extollit.olivine.runtime.attributes.IMobAttribs;
import com.extollit.olivine.runtime.attributes.IMobModifiers;
import com.extollit.olivine.runtime.attributes.IModifier;
import com.extollit.olivine.runtime.attributes.MobProfession;
import com.extollit.olivine.runtime.model.*;
import com.extollit.olivine.runtime.testkit.OlivineRuntimeTestKit;
import com.extollit.olivine.runtime.testkit.TestFixture;
import com.extollit.olivine.runtime.testkit.TestPlayer;
import com.extollit.olivine.runtime.testkit.TestVirtualMachine;
import com.extollit.olivine.runtime.testkit.matchers.Matchers;
import com.extollit.olivine.runtime.testkit.matchers.Vec3dMatcher;
import com.extollit.temporal.Duration;
import org.hamcrest.BaseMatcher;
import org.hamcrest.Description;
import org.junit.Test;
import org.mockito.AdditionalMatchers;
import org.mockito.internal.matchers.EqualsWithDelta;

import java.lang.reflect.InvocationTargetException;
import java.util.Arrays;
import java.util.EnumSet;
import java.util.List;

import static com.extollit.olivine.runtime.testkit.matchers.Matchers.*;
import static org.mockito.Mockito.*;

/**
 * main
 * <p/>
 * Created by jonathan on 05/04/16.
 */
public class DynamicExpressionTests {

    private static final QualifiedReference ANVIL_Sound = new QualifiedReference("anvil");

    private static class TestSpace<ItemAttributes, MobAttributes, FixtureAttributes> extends OlivineRuntimeTestKit {
        @SuppressWarnings("unchecked")
        public final IVirtualMachine<ItemAttributes, MobAttributes, FixtureAttributes, ?> mockVM = mock(IVirtualMachine.class);
        public final TestVirtualMachine<ItemAttributes, MobAttributes, FixtureAttributes, ?> testVM = TestVirtualMachine.wrap(mockVM);
        public final TestPlayer player = new TestPlayer("TheCarnivore", EnumSet.of(Species.human), new Vec3d(1, 2, 3));

        public TestSpace() throws MalformedMorphException, InvalidMorphException, MorphExistsException {
            super.morphRegistry.registerBuiltIn();
            super.gameRegistry.registerItems(new QualifiedReference("foogame", "test_item"));
            super.gameRegistry.registerMobs(new QualifiedReference("foogame", "Villager"));
            super.gameRegistry.registerSounds(ANVIL_Sound);
            testVM.entity(this.player);
        }

        public ICompiledProgram compile(OlivineProgram program) throws ClassNotFoundException, InvocationTargetException, InstantiationException, NoSuchMethodException, IllegalAccessException, OlivineException {
            return super.compile(program, this.testVM, this.player);
        }
        public ICompiledProgram compile(OlivineProgram program, IObject context) throws ClassNotFoundException, InvocationTargetException, InstantiationException, NoSuchMethodException, IllegalAccessException, OlivineException {
            return super.compile(program, this.testVM, context);
        }
        public ICompiledProgram compile(OlivineProgram program, Iterable<? extends IObject> context) throws ClassNotFoundException, InvocationTargetException, InstantiationException, NoSuchMethodException, IllegalAccessException, OlivineException {
            return super.compile(program, this.testVM, context);
        }

        public void register(Class<?> clazz) throws InvalidMorphException, MorphExistsException, MalformedMorphException {
            super.morphRegistry.register(clazz);
        }
    }

    private <ItemAttributes, MobAttributes, FixtureAttributes> TestSpace<ItemAttributes, MobAttributes, FixtureAttributes> space(Class<?> interfaze) throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<ItemAttributes, MobAttributes, FixtureAttributes> testSpace = new TestSpace<ItemAttributes, MobAttributes, FixtureAttributes>();
        testSpace.register(interfaze);

        return testSpace;
    }

    private <ItemAttributes, MobAttributes, FixtureAttributes> void test(String script, TestSpace<ItemAttributes, MobAttributes, FixtureAttributes> testSpace, Object... logging) throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        OlivineProgram program = testSpace.parse(script);
        ICompiledProgram instance = testSpace.compile(program);

        instance.runCycle();
        verify(testSpace.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                eq(logging)
        );
    }

    @Test
    public void add2Vecs () throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/linalg/add-2-vecs.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM).play(eq(ANVIL_Sound), veq(new Vec3d(6, 9, 5)), anyDouble(), anyDouble());
    }

    @Test
    public void sub2Vecs () throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/sub-2-vecs.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM).play(eq(ANVIL_Sound), veq(new Vec3d(4, 5, -1)), anyDouble(), anyDouble());
    }

    @Test
    public void addThenSubVecs () throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/linalg/add-then-sub-vecs.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM).play(eq(ANVIL_Sound), veq(new Vec3d(-1, 5, -4)), anyDouble(), anyDouble());
    }

    @Test
    public void dotProduct () throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/linalg/dot-product.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                AdditionalMatchers.eq(-3.0, 0.5)
        );
    }

    @Test
    public void scaleUpVec () throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/linalg/scale-up-vec.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM, times(2)).play(eq(ANVIL_Sound), veq(new Vec3d(0, 42, 0)), anyDouble(), anyDouble());
        verify(space.mockVM, times(4)).play(eq(ANVIL_Sound), veq(new Vec3d(-42, 42, 0)), anyDouble(), anyDouble());
    }

    @Test
    public void crossProduct () throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/linalg/cross-product.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM).play(eq(ANVIL_Sound), veq(new Vec3d(3, 0, -1)), anyDouble(), anyDouble());
        verify(space.mockVM).play(eq(ANVIL_Sound), veq(new Vec3d(-3, 0, 1)), anyDouble(), anyDouble());
        verify(space.mockVM).play(eq(ANVIL_Sound), veq(new Vec3d(8, 0, -3)), anyDouble(), anyDouble());
        verify(space.mockVM).play(eq(ANVIL_Sound), veq(new Vec3d(-8, 0, 3)), anyDouble(), anyDouble());
    }

    @OlivineType(name = "content", kind = OlivineType.Kind.data)
    public interface IContent {
        int id();
        String name();
    }

    @Test
    public void unstableGuestSpawn() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, IMobAttribs, Void> space = new TestSpace<Void, IMobAttribs, Void>();

        space.register(IMobAttribs.class);
        space.testVM.randomSequence(0, 0, 0, 0.5f);

        OlivineProgram program = space.parse("validation/attributes/mob/unstable-guest-spawn.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();

        verify(space.mockVM).spawn(
            argThat(
                new BaseMatcher<MobDefinition<IMobAttribs>>() {
                    private final int owly = 56;
                    private final MobProfession profession = MobProfession.farmer;

                    @Override
                    public void describeTo(Description description) {
                        description.appendText("owly → ");
                        description.appendValue(this.owly);
                        description.appendText(" && ");
                        description.appendText("profession → ");
                        description.appendValue(this.profession);
                    }

                    @Override
                    public boolean matches(Object o) {
                        if (o instanceof MobDefinition<?>) {
                            @SuppressWarnings("unchecked")
                            MobDefinition<IMobAttribs> mob = (MobDefinition<IMobAttribs>) o;

                            IMobAttribs attributes = mob.attributes;

                            if (attributes != null) {
                                IMobModifiers modifiers = attributes.modifiers();

                                if (modifiers != null) {
                                    IModifier owly = modifiers.owly();

                                    if (owly != null)
                                        return
                                                this.profession.equals(attributes.profession()) &&
                                                this.owly == owly.amount();
                                }
                            }
                        }

                        return false;
                    }
                }
            ),
            org.mockito.Matchers.<Vec3d>any()
        );
    }

    @Test
    public void unstableGenerator() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, IMobAttribs, Void> space = new TestSpace<Void, IMobAttribs, Void>();

        space.testVM.randomSequence(
                // Duplicated due to the nature of the random generator collection
                0.1f, 0.1f, 0.9f, 0.1f, 0.9f,
                0.1f, 0.1f, 0.9f, 0.1f, 0.9f
        );

        OlivineProgram program = space.parse("validation/expressions/unstable-generator.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();

        verify(space.mockVM).log(
            org.mockito.Matchers.<List<String>>any(),
            collection(Arrays.asList("foo", "foo", "gong", "foo", "gong"))
        );
    }

    @Test
    public void negatory() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, IMobAttribs, Void> space = new TestSpace<Void, IMobAttribs, Void>();

        OlivineProgram program = space.parse("compile/expression/negatory.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();

        verify(space.mockVM, times(2)).log(
            org.mockito.Matchers.<List<String>>any(),
            eq(-42), AdditionalMatchers.eq(-3.5, 0.01), eq(Duration.of(-5, Duration.Units.seconds)), veq(new Vec3d(-8, -4, -3))
        );
        verify(space.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                veq(new Vec3d(-8, -4, -3))
        );
    }

    @Test
    public void normalizeVec() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/linalg/normalize-vec.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();

        verify(space.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                veq(new Vec3d(0.566, 0.793, 0.226)),
                veq(new Vec3d(0.619, 0.722, 0.309))
        );
    }

    @Test
    public void mags() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/mags.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();

        verify(space.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                eq(8.831760866327848),
                eq(5)
        );
    }

    @Test
    public void tuples() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/tuples.olivine");
        final Vec3d
                firstCoords = new Vec3d(4, 9, 10),
                secondCoords = new Vec3d(9, 16, -5);
        ICompiledProgram instance = space.compile(program, Arrays.asList(new TestFixture(firstCoords), new TestFixture(secondCoords)));

        space.testVM.randomSequence(0);

        instance.runCycle();

        verify(space.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                eq("foobar"), eq(42), eq(Duration.of(29, Duration.Units.seconds)),
                argThat(
                    lists(
                        Matchers.tuples(new Vec3dMatcher(firstCoords), new EqualsWithDelta(10.34, 0.05)),
                        Matchers.tuples(new Vec3dMatcher(secondCoords), new EqualsWithDelta(18, 0.05))
                    )
                )
        );
    }

    @Test
    public void vecProj () throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("compile/expression/linalg/vec-proj.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                veq(0, 45, 45),
                argThat(lists(new Vec3dMatcher(new Vec3d(1.44554, 2.89109, 6.50495), 0.05f))),
                argThat(lists(new Vec3dMatcher(new Vec3d(0.953728, -0.544987, 2.452442), 0.5f)))
        );
    }

    @Test
    public void operatorStabilityConsistency() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        OlivineProgram program = space.parse("validation/consts/operator-stability-consistency.olivine");
        ICompiledProgram instance = space.compile(program);

        instance.runCycle();
        verify(space.mockVM).log(
                org.mockito.Matchers.<List<String>>any(),
                collection(space.player),
                eq(true)
        );
    }

    @OlivineType(name = "xliving", kind = OlivineType.Kind.contract)
    public interface IExtLivingBeing extends ILivingBeing {
        int foo();
    }

    @Test
    public void lifeExtendPolymorph() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        class ExtPlayer extends TestPlayer<Void> implements IExtLivingBeing {
            public ExtPlayer(String name, Vec3d coordinates) {
                super(name, EnumSet.of(Species.human), coordinates);
            }

            @Override
            public int foo() {
                return 42;
            }
        }

        space.morphRegistry.register(IExtLivingBeing.class);

        ExtPlayer player = new ExtPlayer("bob", new Vec3d(1, 2, 3));

        final OlivineProgram program = space.parse("compile/expression/life-extend-polymorph.olivine");
        final ICompiledProgram instance = space.compile(program, player);

        instance.runCycle();

        verify(space.mockVM).log(anyListOf(String.class), eq(true));
    }
}
```

Workaround is to disable the Minecraft plugin.  And I'd upgrade my IntelliJ except that the latest version has new issues that prevent me from running the tests.

**Comment by DenWav** [comment]
Can you include the stacktrace too?

**Comment by DenWav** [comment]
Also, if this is an open source project, a link to the repo would be helpful, since then I could just open the project to reproduce it.

**Comment by MadMartian** [comment]
It is not open source, but I managed to procure a stack trace:

```
In file: file:///home/me/im/not/that/stupid/to/reveal/local/paths/online/main/olivine/src/test/java/com/extollit/olivine/runtime/DynamicExpressionTests.java: Failed to determine anonymous class for PsiClass:ExtPlayer
java.lang.IllegalStateException: Failed to determine anonymous class for PsiClass:ExtPlayer
	at com.demonwav.mcdev.util.Class_utilsKt.getAnonymousIndex(class-utils.kt:153)
	at com.demonwav.mcdev.util.Class_utilsKt.buildQualifiedName(class-utils.kt:262)
	at com.demonwav.mcdev.util.Class_utilsKt.getFullQualifiedName(class-utils.kt:46)
	at com.demonwav.mcdev.util.MemberReferenceKt.getQualifiedMemberReference(MemberReference.kt:128)
	at com.demonwav.mcdev.util.MemberReferenceKt.isSameReference(MemberReference.kt:136)
	at com.demonwav.mcdev.util.Call_utilsKt.getCallsReturningResult(call-utils.kt:77)
	at com.demonwav.mcdev.i18n.translations.TranslationFunction.getCalls(TranslationFunction.kt:54)
	at com.demonwav.mcdev.i18n.translations.TranslationFunction.matches(TranslationFunction.kt:47)
	at com.demonwav.mcdev.i18n.translations.identifiers.TranslationIdentifier$Companion.identify(TranslationIdentifier.kt:47)
	at com.demonwav.mcdev.i18n.translations.identifiers.LiteralTranslationIdentifier.identify(LiteralTranslationIdentifier.kt:21)
	at com.demonwav.mcdev.i18n.translations.identifiers.LiteralTranslationIdentifier.identify(LiteralTranslationIdentifier.kt:17)
	at com.demonwav.mcdev.i18n.translations.identifiers.TranslationIdentifier.identifyUnsafe(TranslationIdentifier.kt:29)
	at com.demonwav.mcdev.i18n.translations.Translation$Companion$registerReferenceProviders$1.getReferencesByElement(Translation.kt:141)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.getReferences(ReferenceProvidersRegistryImpl.java:135)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.mapNotEmptyReferencesFromProviders(ReferenceProvidersRegistryImpl.java:123)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.doGetReferencesFromProviders(ReferenceProvidersRegistryImpl.java:102)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:59)
	at com.intellij.psi.PsiReferenceServiceImpl.getReferences(PsiReferenceServiceImpl.java:32)
	at com.intellij.psi.PsiReferenceService.getContributedReferences(PsiReferenceService.java:52)
	at com.intellij.psi.impl.source.tree.java.PsiLiteralExpressionImpl.getReferences(PsiLiteralExpressionImpl.java:206)
	at com.intellij.codeInsight.daemon.impl.analysis.HighlightVisitorImpl.visitLiteralExpression(HighlightVisitorImpl.java:809)
	at com.intellij.psi.impl.source.tree.java.PsiLiteralExpressionImpl.accept(PsiLiteralExpressionImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.analysis.HighlightVisitorImpl.visit(HighlightVisitorImpl.java:138)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:371)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$collectHighlights$5(GeneralHighlightingPass.java:309)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:330)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$6(GeneralHighlightingPass.java:333)
	at com.intellij.codeInsight.daemon.impl.analysis.HighlightVisitorImpl.lambda$analyze$2(HighlightVisitorImpl.java:163)
	at com.intellij.codeInsight.daemon.impl.analysis.RefCountHolder.analyze(RefCountHolder.java:336)
	at com.intellij.codeInsight.daemon.impl.analysis.HighlightVisitorImpl.analyze(HighlightVisitorImpl.java:162)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:333)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$6(GeneralHighlightingPass.java:333)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:86)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:333)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:300)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:239)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:83)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:437)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1127)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:430)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:429)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:405)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:143)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:229)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:403)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:170)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by MadMartian** [comment]
This looks implicating, around line 371:

```
    @Test
    public void lifeExtendPolymorph() throws OlivineException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
        TestSpace<Void, Void, Void> space = new TestSpace<Void, Void, Void>();

        class ExtPlayer extends TestPlayer<Void> implements IExtLivingBeing {
            public ExtPlayer(String name, Vec3d coordinates) {
                super(name, EnumSet.of(Species.human), coordinates);
            }

            @Override
            public int foo() {
                return 42;
            }
        }

        space.morphRegistry.register(IExtLivingBeing.class);

        ExtPlayer player = new ExtPlayer("bob", new Vec3d(1, 2, 3));

        final OlivineProgram program = space.parse("compile/expression/life-extend-polymorph.olivine");
        final ICompiledProgram instance = space.compile(program, player);

        instance.runCycle();

        verify(space.mockVM).log(anyListOf(String.class), eq(true));
    }
```
Class `ExtPlayer` is a class nested within a function body.

**Comment by DenWav** [comment]
Thanks for the stacktrace. This is fixed in 17692b54398cf3fdd227cb424befced464832eeb, which is in `2018.1-1.2.8`, `2018.2-1.2.8`, and `2018.3-1.2.8`. Unfortunately a fix for that isn't available for `2017.3`, since I found supporting older versions was rather difficult. Since this is a rather significant bug, I'll see if I can cherry-pick that commit back onto the `2017.3` branch. I'll also do some more research into finding perhaps better methods of supporting larger version gaps.
